### PR TITLE
enhancement/1581 - Add Ability to Reset Scratchpad

### LINKED ISSUE #1581: Add way to clear scratchpad
Add a case where setting a scratchpad to something like "---" would clear the aircraft's scratchpad.

### PR DESCRIPTION
**Pull request by j-651**
Resolves #1581.
Replaces #1592.

Add the ability to clear/reset an aircraft's scratchpad. Typing `. [CID]` (scope command) will reset the aircraft's scratchpad to its destination or first fix as automated on spawn.

### CODE REVIEW

File: src/assets/scripts/client/scope/RadarTargetModel.js [hunk 1]
```diff
@@ -544,6 +526,26 @@ export default class RadarTargetModel {
...
+     * @chainable
+     */
+    setDefaultScratchpad(aircraftModel) {
+        if (aircraftModel.isDeparture()) {
+            this.scratchPadText = aircraftModel.fms.getFlightPlanEntry();
+
+            return [true, 'RESET SCRATCHPAD'];
+        }
+
+        this.scratchPadText = aircraftModel.destination.substr(1, 3);
```
Comment (erikquinn) [inline src/assets/scripts/client/scope/RadarTargetModel.js:544]: We really don't need to _pass_ the aircraftModel like this, we should instead just access it with `this.aircraftModel`, as used in other methods of this class.

This change is the reason some tests are now failing-- because the tests aren't passing the aircraftModel, they're expecting you to retrieve it from the RadarTargetModel.aircraftModel.

File: src/assets/scripts/client/scope/RadarTargetModel.js [hunk 2]
```diff
@@ -172,8 +172,8 @@ export default class RadarTargetModel {
         this._theme = theme;
 
         this._init(aircraftModel)
-            ._initializeScratchPad()
-            .enable();
+            .enable()
+            .setDefaultScratchpad(aircraftModel);
```
Comment (erikquinn) [inline src/assets/scripts/client/scope/RadarTargetModel.js:176]: enable should be done last, only when the thing is fully ready-- so let's reverse these.

File: src/assets/scripts/client/scope/RadarTargetModel.js [hunk 3]
```diff
@@ -299,6 +279,8 @@ export default class RadarTargetModel {
     */
     enable() {
         this._eventBus.on(EVENT.SET_THEME, this._setTheme);
+
+        return this;
```
Comment (erikquinn) [inline src/assets/scripts/client/scope/RadarTargetModel.js:283]: 👍 just add `@chainable` to the docblock to whenever we return `this`.

File: src/assets/scripts/client/scope/ScopeModel.js [hunk 4]
```diff
@@ -302,6 +302,10 @@ export default class ScopeModel {
             return [false, 'ERR: SCRATCHPAD MAX 3 CHAR'];
         }
 
+        if (scratchPadText === '.') {
+            return radarTargetModel.setDefaultScratchpad(radarTargetModel.aircraftModel);
+        }
```
Comment (erikquinn) [inline src/assets/scripts/client/scope/ScopeModel.js:307]: Don't need to pass anything here, since the RadarTargetModel can access its own aircraftModel.

File: src/assets/scripts/client/scope/RadarTargetModel.js (hunk 3 above)
Comment (j-651) [reply src/assets/scripts/client/scope/RadarTargetModel.js:283]: If enable() comes at the end, it shouldn't need to be chainable, right?

File: src/assets/scripts/client/scope/RadarTargetModel.js (hunk 2 above)
Comment (erikquinn) [reply src/assets/scripts/client/scope/RadarTargetModel.js:176]: More accurately (as discussed), we should just do the .setDefaultScratchpad() from _within_ `this._init()`.

That would leave _this_ line as:
```
this._init(aircraftModel)
    .enable();
```

File: src/assets/scripts/client/scope/RadarTargetModel.js (hunk 3 above)
Comment (erikquinn) [reply src/assets/scripts/client/scope/RadarTargetModel.js:283]: all the stuff where we initialize is fine to be chainable (so .init(), .reset(), .enable(), .disable(), ._initSomethingInParticular(), etc). Normal methods like .doSomething() should not be chainable.

File: src/assets/scripts/client/scope/RadarTargetModel.js [hunk 5]
```diff
@@ -544,6 +534,26 @@ export default class RadarTargetModel {
...
 
+    /**
+     * Set default value of the scratchpad
+     *
+     * @for RadarTargetModel
+     * @method setDefaultScratchpad
+     * @private
+     * @chainable
+     */
+    setDefaultScratchpad() {
```
Comment (erikquinn) [inline src/assets/scripts/client/scope/RadarTargetModel.js:545]: This is actually a _public_ method, meaning it can be accessed from OUTSIDE the RadarTargetModel class (which we do-- the ScopeModel calls RadarTargetModel.setDefaultScratchpad). A private method would be one that is only allowed to be accessed from within the RadarTargetModel class. If no other classes _need_ to use a given method, it should be made private (because bugs).

Also, all private methods should begin with an underscore. Since this doesn't have one, that's how we know whether it's public/private at a glance when we go to use it. So if you ever find yourself in a situation where you're doing like `this.someObject._doSomething()`, you _know_ the designer of `someObject` has a reason why you probably shouldn't do that.

tldr: remove `@private`

Comment (j-651) [reply src/assets/scripts/client/scope/RadarTargetModel.js:545]: OK, good to know. That was leftover from when I copy-pasted the `_initializeScratchpad()` function and I didn't make that distinction with underscores and private functions. Pushing correction now.

Comment (j-651) [reply src/assets/scripts/client/scope/RadarTargetModel.js:545]: As I look back at this, I'm also removing `@chainable` since per our conversation that no longer applies and adding `@return {array} [success of operation, system's response]`. since that was missing
